Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,237 @@
+public final class ViewAnnotationManager {
+
+    private let containerView: SubviewInteractionOnlyView
+    private let mapboxMap: MapboxMapProtocol
+    private var annotationViewsById: [String: AnnotationView] = [:]
+    private var annotationIdByViews: [AnnotationView: String] = [:]
+
+    internal init(containerView: SubviewInteractionOnlyView, mapboxMap: MapboxMapProtocol) {
+        self.containerView = containerView
+        self.mapboxMap = mapboxMap
+        let delegatingPositionsListener = DelegatingViewAnnotationPositionsUpdateListener()
+        delegatingPositionsListener.delegate = self
+        mapboxMap.setViewAnnotationPositionsUpdateListener(delegatingPositionsListener)
+    }
+    
+    deinit {
+        mapboxMap.setViewAnnotationPositionsUpdateListener(nil)
+    }
+
+    // MARK: - Public APIs
+
+    /// Add annotation to the map which wraps a supplied `UIView`
+    /// View dimensions will be taken as width / height from the bounds of the view
+    /// unless they are not specified explicitly with `ViewAnnotationOptions.width` and `ViewAnnotationOptions.height`.
+    ///
+    /// Annotation `options` must include Geometry where we want to bind our view annotation.
+    ///
+    /// Width and height could be specified explicitly but better idea will be not specifying them
+    /// as they will be calculated automatically based on view layout.
+    ///
+    /// - Parameters:
+    ///   - view: `UIView` to be wrapped in an `AnnotationView` and placed on the map
+    ///   - options: view annotation options
+    ///
+    /// - Throws: `ViewAnnotationManagerError.insertionFailure` if options did not include geometry or contains incomplatible values.
+    public func addAnnotationView(withContent view: UIView, options: ViewAnnotationOptions) throws -> AnnotationView {
+        guard options.geometry != nil else {
+            throw ViewAnnotationManagerError.insertionFailure(reason: "Geometry field is missing")
+        }
+        var creationOptions = options
+        if creationOptions.width == nil {
+            creationOptions.width = view.bounds.size.width
+        }
+        if creationOptions.height == nil {
+            creationOptions.height = view.bounds.size.height
+        }
+        let annotatonView = AnnotationView(view: view, annotationManager: self)
+        try mapboxMap.addViewAnnotation(withId: annotatonView.id, options: options)
+        annotationViewsById[annotatonView.id] = annotatonView
+        annotationIdByViews[annotatonView] = annotatonView.id
+        containerView.addSubview(annotatonView)
+        return annotatonView
+    }
+
+    /// Remove given `AnnotationView` from the map if it was present.
+    ///
+    /// - Throws:
+    ///   - `ViewAnnotationManagerError.annotationNotFound`: the supplioed view was not found
+    ///   - `ViewAnnotationManagerError.removalFailure`: errors during the removal of the view
+    public func remove(_ annotatonView: AnnotationView) throws {
```

Comment:
I've removed the boolean return value from this function and from `update` because we have to throw errors from the GL-Native side and I thought it might be more convenient for the user to handle the "not found" case the same way as other issues. (I can of course put back the return value if needed, no strong opinions on that.)

Reply:
Agreed. Returning a Bool would be redundant